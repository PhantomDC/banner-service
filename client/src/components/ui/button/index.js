import React, { Component } from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'


const Button = ({ className, text, ...rest }) => (
  <button className={className} {...rest}>
    {text}
  </button>
)

const btnWithStyle = styled(Button)`
  outline: none;
  border: none;
  text-align: center;
  padding: 10px 15px;
  cursor: pointer
`

export default btnWithStyle
