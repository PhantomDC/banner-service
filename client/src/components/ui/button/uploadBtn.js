import React, { Component } from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import { Button } from '../index'

class uploadBtn extends Component {
  componentDidMount() {
    this.refs.files.addEventListener('change', this.props._changeInput)
  }

  componentWillUnmount() {
    this.refs.files.removeEventListener('change', this.props._changeInput)
  }

  _handleInputFile = () => {
    this.refs.files.click()
  }

  render() {
    const { className, text, _changeInput, ...rest } = this.props
    return (
      <div>
        <input type="file" ref="files" />
        <Button
          onClick={this._handleInputFile}
          text={text}
          {...rest}
        />
      </div>
    )
  }
}

uploadBtn.propTypes = {
  text: PropTypes.string.isRequired,
  _changeInput: PropTypes.func.isRequired
}

export default uploadBtn
