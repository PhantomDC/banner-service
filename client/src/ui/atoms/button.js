import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'


const btn = ({ className, text, ...rest }) => (
  <button className={className} {...rest}>
    {text}
  </button>
)

export const Button = styled(btn)`
  outline: none;
  border: none;
  text-align: center;
  padding: 10px 15px;
  cursor: pointer
`
