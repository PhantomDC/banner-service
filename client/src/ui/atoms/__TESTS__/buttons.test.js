import React from 'react'
import { configure, render, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { create } from 'react-test-renderer'
import { Button } from '../button'

/* eslint-disable no-undef */
/* eslint-disable function-paren-newline */

configure({ adapter: new Adapter() })

describe('test component Button', () => {
  it('Button render without crash', () => {
    render(<Button text="upload" />)
  })

  it('text props test', () => {
    const wrapper = shallow(<Button text="upload" />)
    const { text } = wrapper.instance().props

    expect(text).toBe('upload')
  })

  it('shapshot Button', () => {
    const tree = create(<Button text="upload" />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
