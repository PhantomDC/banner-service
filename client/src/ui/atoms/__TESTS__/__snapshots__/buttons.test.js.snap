// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component Button shapshot Button 1`] = `
<button
  className="sc-bdVaJa hvseiO"
>
  upload
</button>
`;
